piecedirectory/piecedirectory.go

```diff
@@ -222,17 +248,44 @@ func (ps *PieceDirectory) addIndexForPiece(ctx context.Context, pieceCid cid.Cid
 		blockMetadata, err = blockReader.SkipNext()
 	}
 	if !errors.Is(err, io.EOF) {
-		return fmt.Errorf("generating index for piece %s: %w", pieceCid, err)
+		return nil, fmt.Errorf("generating index for piece: %w", err)
 	}
+	return recs, nil
+}
 
-	// Add mh => piece index to store: "which piece contains the multihash?"
-	// Add mh => offset index to store: "what is the offset of the multihash within the piece?"
-	log.Debugw("add index: store index in local index directory", "pieceCid", pieceCid)
-	if err := ps.store.AddIndex(ctx, pieceCid, recs, true); err != nil {
-		return fmt.Errorf("adding CAR index for piece %s: %w", pieceCid, err)
+func parseShardWithDataSegmentIndex(ctx context.Context, pieceCid cid.Cid, unpaddedSize int64, r types.SectionReader) ([]model.Record, error) {
```

Comment:
nit: I suggest we use the word "piece" instead of shard. Shard only has meaning in the dagstore context